core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -709,6 +717,8 @@ func (fm *FluxMonitor) respondToNewRoundLog(log flux_aggregator_wrapper.FluxAggr
 		}
 		return fm.logBroadcaster.MarkConsumed(tx, lb)
```

Comment:
hm maybe a bit more clear would be to remove this one and have MarkConsumed only in defer? 
You could put `markConsumed = false` inside the if

Reply:
We'd then lose the transactionality that we want though. We only want to mark consumed outside of a tx if we don't actually do a run

Reply:
oh true, indeed